### PR TITLE
add main deploy

### PR DESCRIPTION
**Pull request by logonoff**
Using the types package

### CODE REVIEW

File: .github/workflows/devdeploy.yml [hunk 1]
```diff
@@ -18,4 +18,4 @@ jobs:
       - name: cleanup
         run: |
           docker image prune -f
-          docker builder prune -f
+          docker builder prune -f
```
Comment (ggggg) [inline .github/workflows/devdeploy.yml:21]: This is just a white space change? Shouldn't we keep the trailing line anyhow?

File: frontend/components/AssignmentCard/AssignmentCard.tsx [hunk 2]
```diff
@@ -1,8 +1,7 @@
-import { Caption1, Card, CardHeader, Title3 } from '@/components/fluent';
-import Link from 'next/link';
+import { Caption1, Card, CardHeader, Title3, Link } from '@fluentui/react-components';
 import styles from './AssignmentCard.module.css';
-import { TierChip } from '@/components/client';
-import { type DisplayTier, type Tier } from '@/lib/types';
+import { TierChip } from '../TierChip/TierChip';
+import { type DisplayTier, type Tier } from '../../../backend/src/types/global';
```
Comment (ggggg) [inline frontend/components/AssignmentCard/AssignmentCard.tsx:4]: Not how types are handled, import from type package

**Review by Yousef-Bulbulia — APPROVED** [review]

File: frontend/components/Navbar/Navbar.tsx [hunk 3]
```diff
@@ -1,16 +1,14 @@
...
+import { GenerateInitalsAvatarProps } from '@/components';
 
 /**
  * The navbar component is the top bar of the website. It contains the
  * logo and the user's name and role.
  * @returns {JSX.Element} the navbar
  */
 export const Navbar = (): JSX.Element => {
-    const user = { utorid: getUtorid(), email: getEmail() };
+    const user = { utorid: "michael", email: "liut" };
```
Comment (ggggg) [inline frontend/components/Navbar/Navbar.tsx:11]: Assuming this is mock data for now, should add a central function like `getUser` so it is easier to integrate in the future

File: frontend/tsconfig.json [hunk 4]
```diff
@@ -19,9 +19,9 @@
       }
     ],
     "paths": {
-      "@/*": ["./src/*"]
+      "@/*": ["./*"]
     }
   },
-  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts"],
+  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts", "pages/courses/index.tsx", "../frontend/pages/404.tsx"],
```
Comment (ggggg) [inline frontend/tsconfig.json:25]: Not sure why pages need to be included explicitly

**Review by ggggg — CHANGES_REQUESTED** [review]

**Review by ggggg — APPROVED** [review]

File: frontend/pages/courses/[courseID]/[assignmentID]/page.module.css
Comment (ggggg) [inline frontend/pages/courses/[courseID]/[assignmentID]/page.module.css:1]: Empty file?

**Pull request merged**